src/pools/thread/fixed.ts

```diff
@@ -80,13 +96,34 @@ export class FixedThreadPool<
     })
   }
 
+  protected internalExecute (
+    worker: Worker,
+    messageId: number
+  ): Promise<Response> {
+    return new Promise<Response>((resolve, reject) => {
+      this.promiseMap.set(messageId, { resolve, reject, worker })
+    })
+  }
+
   protected afterWorkerSetup (worker: ThreadWorkerWithMessageChannel): void {
     const { port1, port2 } = new MessageChannel()
     worker.postMessage({ parent: port1 }, [port1])
     worker.port1 = port1
     worker.port2 = port2
     // We will attach a listener for every task,
-    // when task is completed the listener will be removed but to avoid warnings we are increasing the max listeners size
+    // when the task is completed the listener will be removed but to avoid warnings we are increasing the max listeners size.
     worker.port2.setMaxListeners(this.opts.maxTasks ?? 1000)
+    const listener: (message: MessageValue<Response>) => void = message => {
```

Comment:
Is this only available in thread pools? If so: why?

Reply:
i added this only to threads for simplicity, if you and @jerome-benoit do not see drawbacks I would add It directly to the abstract pool

Reply:
I cannot say too much to this because I'm not so deep in the technical aspects but more into the quality etc. ^^'
So @jerome-benoit needs to look into this

Reply:
@jerome-benoit I would try ti add It to the abstract pool of you agree

Reply:
I fully agree. It's a global performance optimisation.

Reply:
Done